### ISSUE TITLE
securefs support

### ISSUE THREAD

**Issue by bingoxo** [hardcore-sushi/DroidFS#4]
Hi , according to https://nuetzlich.net/gocryptfs/comparison/ , securefs seems to be better performing than cryfs , can you look into it please ? https://github.com/netheril96/securefs

**Comment by hardcore-sushi** [comment]
Yes indeed, it seems to be faster. However, (according to this same webpage) it appears to be less secure than CryFS.
The main goal of DroidFS being security, I will keep priority over CryFS.
In the future, I might add securefs support for those who would like more performance at the expense of security.

**Comment by dhammel** [comment]
From what i've researched about securefs, it seems to be more secure than the other alternatives like gocryptfs. Main concern I have is that it doesnt offer filesize Obfuscation, but neither does gocryptfs it seems. Developer of securefs mentioned the reason he cteated it was a priority over security that others didnt offer other than maybe veracrypt, but that isnt optimized for cloud usage. Ive never used gocryptfs until your app, so im still going over its documention and lookjng through its, and your source. Your app however has been what ive been looking for, especially as its open source. securefs however would be a nice addition, but as long as your gocryptfs implementation is secure than I wouldnt say it should be priority, probably best to rely on a platform like android for too mich real sensitive data anyway.

**Comment by hardcore-sushi** [comment]
Yes I agree that securefs seems to be more secure than gocryptfs (if you use it in full mode). I'm not a cryptography expert, but it seems that securefs is however less secure than CryFS because of its KDF function, its lack of file size obfuscation and leaks when file names are identical. But of course I don't rule out the possibility of adding securefs to DroidFS someday.

**Comment by dhammel** [comment]
As for the KDF, foe encryption to be secure its not a rewuirement to even have a KDF as long as thr password is long, secure and random. The fact it at leasg does have one, makes it more secure than wothout for users with weaker passwords. CryFS has file size Obfuscation? I havent done as much research with that program. It might be best to determine what encryption tools to implement and what not too, as to not have such higher maintienence burder or possible flaws come up. Maybe no more than 3 for e.g. I will start looking more into CryFS that, as I currently used srcurefs, it it is more secure I may switch.

**Comment by yashikada** [comment]
Please reopen this FR. Now securefs support also "File size obfuscation by random padding", is optional but is supported.

**Comment by yashikada** [comment]
No reply after about 1 year. 
Now securefs compile also on arm and has the version static. 
I can run on termux without use this proect. Just download and run.
I think is better, so it's integrated on android.

**Comment by gityhubi** [comment]
I agree. Request to the developer. Could you please consider adding securefs to DroidFS?
